File: Back-end/src/main/java/com/service/cms_atn/service/UnitService.java
package com.service.cms_atn.service;

import com.service.cms_atn.model.Unit;
import com.service.cms_atn.response.BaseResponse;

public interface UnitService {

	BaseResponse getAll(Unit unit);

	BaseResponse getById(Unit unit);

	BaseResponse create(Unit unit);

	BaseResponse update(Unit unit);

	BaseResponse delete(Unit unit);
}

File: Font-end/src/app/core/services/unitService.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { CommandURL } from 'src/app/shares/const/ManageURLCommand';

@Injectable({
    providedIn: 'root'
})
export class UnitService {

    constructor(private http: HttpClient) { }

    getAll(json: any) {
        return this.http.post<any>(CommandURL.GET_ALL_UNIT, json);
    }

    create(json: any) {
        return this.http.post<any>(CommandURL.CREATE_UNIT, json);
    }

    update(json: any) {
        return this.http.post<any>(CommandURL.UPDATE_UNIT, json);
    }

    delete(json: any) {
        return this.http.post<any>(CommandURL.DELETE_UNIT, json);
    }

}